app/helpers/random_assign_helper.rb

```diff
@@ -0,0 +1,158 @@
+require 'set'
+
+class UnableToRandomlyAssignGroupException < Exception
+end
+
+module RandomAssignHelper
+  # Performs the random assignment, and stores them in the database.
+  # - pr_assignment: The peer review assignment.
+  # - num_groups_for_reviewers: How many PR's each group in reviewer_groups should have.
+  def perform_random_assignment(pr_assignment, num_groups_for_reviewers)
+    @eligible_reviewers = []
+    @shuffled_reviewees = []
+    @shuffled_reviewees_assigned_reviewers = []
+    @num_groups_for_reviewers = num_groups_for_reviewers
+    @reviewer_ids_assigned_reviewee_ids_map = Hash.new { |h, k| h[k] = Set.new }
+
+    reviewer_groups_relation = pr_assignment.valid_groupings
+    reviewee_groups_relation = pr_assignment.parent_assignment.valid_groupings
+    reviewer_groups_relation.each { |reviewer| @eligible_reviewers.push(reviewer) }
+
+    generate_shuffled_reviewees(reviewer_groups_relation, reviewee_groups_relation)
+    populate_from_existing_peer_reviews(pr_assignment)
+    perform_assignments()
+  end
+
+  private
+  # Make sure there's enough reviewee groups in the following list such that
+  # we have enough repeats of 'reviewee_groups' that we can assign every
+  # reviewer group at least 'num_groups_for_reviewers' groups.
+  def generate_shuffled_reviewees(reviewer_groups, reviewee_groups)
+    num_times_to_add_reviewee = (reviewer_groups.size.to_f * @num_groups_for_reviewers / reviewer_groups.size).ceil
+    num_times_to_add_reviewee.times { @shuffled_reviewees += reviewee_groups }
+    @shuffled_reviewees.shuffle()
+
+    @shuffled_reviewees_assigned_reviewers = @shuffled_reviewees.map { |dummy| nil }
+  end
+
+  # We need to get all the existing peer reviews for assignments so we can
+  # skip some assignments if they exist.
+  def populate_from_existing_peer_reviews(pr_assignment)
+    # TODO : Remove? --- pr_assignment.get_peer_reviews().each do |peer_review|
+    # TODO - Is this okay? the 'peer reviews' forces us to go through the parent... doesn't seem right (do we have a choice?)
+    pr_assignment.parent_assignment.peer_reviews.each do |peer_review|
```

Comment:
It should be possible to do this by defining a properly named `:through` association